### ISSUE TITLE
please tag releases

### ISSUE THREAD

**Issue by umlaeute** [pure-data/pd-lib-builder#28]
having tagged releases makes it much easier to refer to a given version than a commitish.

**Comment by katjav** [comment]
Agree. The first version I tagged is v0.4.0. Was disappointed to see that github doesn't show tag titles in the tag list (unlike the commit list).

What about tagging earlier versions retroactively? Release dates wouldn't make much sense.

**Comment by danomatika** [comment]
FWIW I back-tagged libpd versions for completeness when we started following semantic versioning.

**Comment by katjav** [comment]
@danomatika, how did you? I don't see tags before the start of semantic versioning on https://github.com/libpd/libpd.

**Comment by danomatika** [comment]
You don't? The tags listed go back to 0.1.0...

**Comment by katjav** [comment]
Sorry I didn't get it. Did you (at a later moment than when starting semantic versioning) back-tag some versions that were missing in the tag list? How did you get the release date 'right'? I tagged commit cf04da9 from 12 days ago with 'v0.4.0' today, and tagging date == release date. So if I back-tag older commits in a similar way it will make a weird release history.

**Comment by umlaeute** [comment]
i think you only need to tag them in the correct order (e.g. tag `0.0.1` before tagging `0.4.0`), so the list on the [releases/page](../releases) sorts correctly.

you can always just delete a tag (`git push origin :tagname`), then (re)tag all the releases in the correct order and push that.

**Comment by danomatika** [comment]
@umlaeute yup, I didn't worry about the dates

**Comment by katjav** [comment]
The tags for libpd are in plausible chronological order, not like two or more of them were done on the same day. Anyway.

**Comment by danomatika** [comment]
I just checkout a previous commit, tagged, and pushed the tag, then repeated. That was after the several hours of going through the commit history to determine where the natural versions fell. :)

**Comment by residuum** [comment]
This question on stackoverflow is relevant to this discussion: http://stackoverflow.com/questions/21738647/change-date-of-git-tag-or-github-release-based-on-it

**Comment by danomatika** [comment]
Funny, I didn't have to do any of that and the dates are correct for libpd. Just did a checkout and tag.

**Comment by umlaeute** [comment]
i woudln't worry too much about those dates.
i think having proper tags is more important than the sort order on github, esp. since most people will only ever be interested in the latest tag which will show up on top anyhow (provided that you at least one release after the tagging-the-old session).

**Comment by katjav** [comment]
Thanks for the pointer @residuum, showing how to fabricate a release date.

Think I'll first do the archeology and make a brief changelog text with version/date/commit/change info. Based on that the tags for older versions can be fabricated. If most people will indeed use the latest, this is not a priority. I'll try to consistently tag versions from now on. As it seems that git submodules can also point to a version tag, it is extra useful to be serious with it.

**Comment by katjav** [comment]
Added CHANGELOG.txt with commit 006d957 and completed tags for all versions. For the record, I used this to set the time stamp for each tag (based on info in the thread referenced by @residuum):

`GIT_COMMITTER_DATE="$(git show <SHA> --format=%aD | head -1)" git tag -a <version> <SHA>`

Lightweight tags wouldn't need this because they carry only the date of original commit. But I wanted annotated tags.
